sources/packages/api/src/services/student/student.service.ts

```diff
@@ -174,44 +186,102 @@ export class StudentService extends RecordDataModelService<Student> {
    * @returns Students pending SIN validation.
    */
   async getStudentsPendingSinValidation(): Promise<Student[]> {
-    return this.repo.find({ validSIN: null });
+    return this.repo
+      .createQueryBuilder("student")
+      .select([
+        "student.id",
+        "student.sin",
+        "student.birthDate",
+        "user.firstName",
+        "user.lastName",
+        "user.id",
+        "sinValidations.id",
+      ])
+      .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidations")
+      .where("sinValidations.isValidSIN is null")
+      .andWhere("sinValidations.dateSent is null")
+      .andWhere("sinValidations.dateReceived is null")
+      .getMany();
   }
 
   /**
    * Gets a student's valid sin status
    * @returns Students SIN validation status.
    * @param userId information needed to select the user.
    */
-  async getStudentSinStatus(userId: number): Promise<Student> {
-    return await this.repo
+  async getStudentSinStatus(userId: number): Promise<boolean> {
+    const student = await this.repo
       .createQueryBuilder("student")
       .innerJoin("student.user", "user")
+      .innerJoin("student.sinValidation", "sinValidation")
```

Comment:
This indicates that once a student completes first time login with SIN and other information, SIN validation table is populated with a record. Isn't it?

Reply:
Yes, the idea is that there will be always a record associated with the student that will store the most recent SIN validation result. Upon student creation and when the student changes his name, last name, DOB, a record will be created on the table sin_validations and associated with the student.